cmd/checkmetrics/README.md

```diff
@@ -0,0 +1,129 @@
+TOML file. Failure to find a matching CSV is classified as a failure.
+
+`checkmetrics` will continue to process all entries in the TOML file
+and print its final results in a summary table to `stdout`.
+
+Each CSV metrics file must contain at least one header line, with the
+`Results` in column 5, and at least one line containing a test result
+line (that is, the minimum CSV file consists of two lines, one header
+and one result value).
+
+`checkmetrics` will exit with a failure code if any of the TOML entries
+do not complete successfully.
+
+## baseline TOML layout
+The Metrics CSV files to be checked, and the range that the median of the
+`Results` data is expected to fall within is defined in a TOML file.
+
+Each Metric has a separate `[[metric]]` section within the TOML file.
+
+A basic TOML file looks like:
+
+
+```
+[[metric]]
+# The name of the metrics test, must match that of the generated CSV file
+name = "docker-run-time"
+description = "/bin/true cycle time"
+# Min and Max values to set a 'range' that the median of the CSV Results data
+# must fall within (inclusive)
+minval = 0.5
+maxval = 1.5
+
+... and repeat for each metric ...
+```
+
+## Options
+`checkmetrics` takes a number of options. Some are mandatory.
+
+### TOML basefile path (mandatory)
+
+```
+--basefile value    path to baseline TOML metrics file
+```
+
+### Debug mode
+```
+--debug             enable debug output in the log
+```
+
+### Log file path
+```
+--log value         set the log file path
+```
+
+### Metrics CSV directory path
+```
+--metricsdir value  directory containing CSV metrics (mandatory)
+```
+
+### Help
+```
+--help, -h          show help
+```
+
+### Version
+```
+--version, -v       print the version
+```
+
+## Output
+The `checkmetrics` tool outputs a summary table after processing all metrics/CSV
+files, and will return a non-zero return code if any of the metrics checks fail.
+
+In the following example the basefile was configured to process the same CSV file
+four times:
+ - One to pass
+ - One to fail due to a minval failure
+ - One to fail due to a maxval failure
+ - One to fail due to a malformed entry (missing CSV file)
+
+Example output.
+
+```
+Report Summary:
++------+-----------------+--------------------+----------------------------+---------+-------+-------+------+
+| P/F  |      NAME       |       FLOOR        |            MEAN            | CEILING | ITERS |  SD   | COV  |
++------+-----------------+--------------------+----------------------------+---------+-------+-------+------+
+| Pass | docker-run-time |              0.500 |                      1.214 |   1.500 |   100 | 0.036 | 2.96 |
+| Fail | docker-run-time |             10.000 |                      1.214 | 100.000 |   100 | 0.036 | 2.96 |
+| Fail | docker-run-time |              0.100 |                      1.214 |   0.500 |   100 | 0.036 | 2.96 |
+| Fail | short           | Failed to load CSV | Not enough records in file |
```

Comment:
Yeah, I know the formatting is out - to do that right I'd have to add some sort of func to the compare/metricsCheck struct to either determine how many columns we need, or have it fill out/generate the line for us (probably preferable). It's a sort of nasty data dependency across modules that I don't like... as a malformed or missing CSV file should be rare, I was going with 'good enough' for now. I'll take a look and see if I can improve it.
The CSV file names are the same as the test name (with .csv extension), so the failing file can be determined from the name of the test that failed (in the example, 'short').